wicket/src/ui/widgets/ignition.rs

```diff
@@ -48,36 +47,41 @@ impl IgnitionPopup {
         };
     }
 
-    pub fn popup(&self, component: ComponentId) -> Popup<'static> {
-        Popup {
-            header: Text::from(vec![Spans::from(vec![Span::styled(
-                format!(" IGNITION: {}", component),
-                style::header(true),
-            )])]),
-            body: Text {
-                lines: vec![
-                    Spans::from(vec![Span::styled(
-                        "Power On",
-                        style::line(
-                            self.selected_command == IgnitionCommand::PowerOn,
-                        ),
-                    )]),
-                    Spans::from(vec![Span::styled(
-                        "Power Off",
-                        style::line(
-                            self.selected_command == IgnitionCommand::PowerOff,
-                        ),
-                    )]),
-                    Spans::from(vec![Span::styled(
-                        "Power Reset",
-                        style::line(
-                            self.selected_command
-                                == IgnitionCommand::PowerReset,
-                        ),
-                    )]),
-                ],
-            },
-            buttons: vec![ButtonText { instruction: "CLOSE", key: "ESC" }],
-        }
+    /// Return the components to create a popup -- the header, body and button
+    /// text.
+    ///
+    /// Can't return a `Popup` here due to lifetime issues.
+    pub fn popup_components(
+        &self,
+        component: ComponentId,
+    ) -> (Spans<'static>, Text<'static>, Vec<ButtonText>) {
```

Comment:
This seems fine since all three elements are different types, but would it be cleaner to have a `PopupComponents` type?

Reply:
I think this is a good idea. What I'm going to do is rename the old `Popup` into `PopupComponents` and make `Popup` derive from that. That should also clean up the old PR a bit.

Reply:
Done, this is a lot cleaner now. Thank you!